rocAL/rocAL/source/coco_file_source_reader.cpp

```diff
@@ -159,14 +222,28 @@ int COCOFileSourceReader::release()
     if (_current_ifs.bad())
         return 0;
     _current_ifs.close();
-#endif    
+#endif
     return 0;
 }
 
 void COCOFileSourceReader::reset()
 {
     if (_shuffle)
-        std::random_shuffle(_file_names.begin(), _file_names.end());
+    {
+        if(_keypoint)
+        {
+            //shuffle both file names and annotations id's in same order
+            auto seed = unsigned (std::time(0));
```

Comment:
Check the comments before for using srand.

Reply:
Done